File: findjudge.swift
// Time Complexity : O(N^3)
// Space Complexity : O(N)
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach
    func findJudge(_ n: Int, _ trust: [[Int]]) -> Int {
        if (trust.count == 0 || trust == nil) && n > 1 {
            return -1
        } else if n == 1 {
            return 1
        }
        var graph = [Int:[Int]]()
        for value in trust {
            if let people = graph[value[0]] {
                var people = people
                people.append(value[1])
                graph[value[0]] = people
            } else {
                graph[value[0]] = [value[1]]
            }
    }
    var count = 0
        var propablejudge = 0
    for person in 1...n {
        if graph[person] == nil  {
            count += 1
            propablejudge = person
        }
    }
    if count >= 2 {
        return -1
    }
    return isJudge(graph,propablejudge) ? propablejudge : -1
    }
    private func isJudge(_ graph: [Int:[Int]], _ person: Int) -> Bool {
        for (_,people) in graph {
            if !people.contains(person) {
                return false
            }
        }
        return true
    }

File: FindJudge.cs

// Time Complexity : On(n)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach


public int FindJudge(int n, int[][] trust) {
        
    int[] indegrees = new int[n + 1];
    foreach(int[] t in trust)
    {
        //For every indreee , add +1
        //For every outdegree element, subtract -1
        indegrees[t[0]]--;
        indegrees[t[1]]++;
    }
    
    //There will be one index equals to n-1, that will be our answer
    for(int i = 1; i <= n; i++)
    {
        if(indegrees[i] == n-1)
            return i;
    }
    return -1;
}